src/NUnitFramework/framework/Internal/Filters/PartitionFilter.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            // Split our numberWithCount into two parts, such that "1/10" becomes PartitionNumber 1, PartitionCount 10
+            string[] parts = value.Split('/');
+
+            // Parts must be exactly 2, and be in the format of "number/count"
+            if (parts.Length == 2 && uint.TryParse(parts[0], out uint number) && uint.TryParse(parts[1], out uint count)) {
+                // Number must be between 1 and Count, inclusive
+                // Return a new PartitionFilter with the parsed values
+                if (number >= 1 && number <= count)
+                {
+                    partitionFilter = new PartitionFilter(number, count);
+                    return true;
+                }
+            }
+
+            // Could not parse partition information
+            partitionFilter = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Match a test against a single value.
+        /// </summary>
+        public override bool Match(ITest test)
+        {
+            // Do not match a test Suite, only match individual tests
+            if (test.IsSuite)
+                return false;
+
+            // Calculate the partition number for the provided Test
+            var partitionForTest = ComputePartitionNumber(test);
+
+            // Return a match if the calculated partition number matches our configured Partition Number
+            return partitionForTest == PartitionNumber;
+        }
+
+        /// <summary>
+        /// Adds a PartitionFilter XML node to the provided parentNode
+        /// </summary>
+        /// <param name="parentNode">Parent node</param>
+        /// <param name="recursive">True if recursive</param>
+        /// <returns>The added XML node</returns>
+        public override TNode AddToXml(TNode parentNode, bool recursive)
+        {
+            return parentNode.AddElement("partition", $"{PartitionNumber}/{PartitionCount}");
+        }
+
+        /// <summary>
+        /// Computes the Partition Number that has been assigned to the provided ITest value (based upon the configured Partition Count)
+        /// </summary>
+        /// <param name="value">A partition value between 1 and PartitionCount, inclusive</param>
+        /// <returns>A partition value between 1 and PartitionCount, inclusive</returns>
+        public uint ComputePartitionNumber(ITest value)
+        {
+            return ComputeHashValue(value.FullName) % PartitionCount + 1;
+        }
+
+        /// <summary>
+        /// Computes an unsigned integer hash value based upon the provided string
+        /// </summary>
+        private static uint ComputeHashValue(string name)
+        {
+            using var hashAlgorithm = SHA256.Create();
+
+            // SHA256 ComputeHash will return 32 bytes, we will use the first 4 bytes of that to convert to an unsigned integer
+            var hashValue = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(name));
```

Comment:
I wonder how expensive this is and if we can not simply use `name.GetHashCode()`

Reply:
That's a good point, and not something I thought of in the PR to 3.14
Thanks for raising it @manfred-brands 

I'm going to merge this in order to sync the changes between the branches but this is certainly a possibility worth a future change

Reply:
I've put together a dotnetfiddle to check, https://dotnetfiddle.net/xvn7zM

Both implementations give roughly the same distribution, speed is faster in GetHashCode by ~19x but trivial in terms of timing (50ms for 1m hashes in GetHashCode, 950ms for SHA256).

While both are evenly distributed, I note the partition assigned via GetHashCode changes on each iteration (hashing the memory address of the string I assume) where-as the SHA256 of the same text is stable between iterations.

Reply:
Thanks @Plasma
For .NET6 and later the performance difference can be limited to a factor 10 when using `var hashValue = SHA256.HashData(Encoding.UTF8.GetBytes(name));` instead.
The same improvement can be obtained by creating the SHA256 instance once instead of for ever test name.

Maybe the changing distribution of `GetHashCode` is a good thing so tests are more randomly placed on different build runners instead of always on the same.

Reply:
Thanks @manfred-brands good to know RE: HashData method for .NET 6 onwards. I didn't create a single field/instance within the method as the attribute isn't `IDisposable` and thought it may be too much of a change.

RE: `GetHashCode` I lean towards having stability between runs (assuming the partition count, test names and NUnit version didn't change of course) as being important, for example, re-trying a failed GH Action job for a partition of tests that failed should re-run the same tests next time (which isn't promised with `GetHashCode`), and we can rely upon the hashing function fairly splitting tests between partitions.